### ISSUE TITLE
NodeJS not reading package configuration json

### ISSUE THREAD

**Issue by AlbertWillemsen-Centric** [systemjs/systemjs#1655]
I have a few sub project with `defaultExtension: "ts"` configured in their respective package.json under "jspm".
I see them appearing in the jspm_packages/*.json.
After i install the package.
But these values are ignored when running under NodeJS.

With Kind Regards,
Albert

**Comment by guybedford** [comment]
If using jspm in NodeJS, it's advisable to use `jspm run main.js` instead of `node main.js` to ensure the exact configuration and loading semantics.

**Comment by AlbertWillemsen-Centric** [comment]
I'm using SystemJS inside Gulp.
Using:
`import { Builder } from "jspm";
let builder = new Builder();
await builder.bundle(bundleExpression, bundlePath, { minify: true })`

**Comment by guybedford** [comment]
That does sound like a bug then. Can you perhaps try to share a small test case (not the whole project), that just isolates the issue? Would it be enough to just have a single package with `defaultExtension: 'ts'` being bundled this way to replicate it?

**Comment by AlbertWillemsen-Centric** [comment]
I'll build a small test case for you.

Some context:
I'm working on a pure TypeScript project at the moment.
Third party modules are bundled, all my modules are transpiled during development for quicker iterations.
For development everything is bundled using JSPM API bundle.

I'm using Handlebars for client and development prototyping.
So many components will be rendered on the client and server using the same JSPM configurations.
So separate browser, dev and node configuration files.

**Comment by guybedford** [comment]
Let me know if you do have an update here though.

**Comment by AlbertWillemsen-Centric** [comment]
I think i pinpointed this issue.
This is what causes issue #1625
I'll submit a pull or example today.

**Comment by AlbertWillemsen-Centric** [comment]
I have attached a test project.
[package-configuration-test.zip](https://github.com/systemjs/systemjs/files/1161863/package-configuration-test.zip)

So I have 3 packages all typescript.
**Server:**
browser-sync, jspm and main.ts
**Core:**
contains widget initialise and base class.
**Component:**
has dependency on Core.
contains implementation of widget.

This issue has 2 parts.
1. As soon as i try to load the widget module it assumes it's a defaultExtension "js".
Even if the defaultExtension on the config is "ts".

2. Also if the following is configured.
```
SystemJS.config({
  packages: {
    "local:component@1.0.0": {
      "map": {
        "core": "local:core@1.0.0"
      }
    }
  }
});

```
Because of this SystemJS thinks local:component@1.0.0 is fully configures and does not add local:component@1.0.0.json to this configuration.

If i leave packages empty it passes.

**Comment by guybedford** [comment]
Can this be closed along with #1625 then?

**Comment by AlbertWillemsen-Centric** [comment]
If it is fixed.

**Comment by AlbertWillemsen-Centric** [comment]
Let me rephrase the issue:
If a "map" is present for a package in jspm.config.js, then isConfigured is set to true and the package's json file is ignored.
So it won't load "main", "defaultExtension" in this json.

I have been debugging, but am unable to pinpoint the issue.
I think it occurs somewhere in the System.config function.

**Comment by guybedford** [comment]
@AlbertWillemsen-Centric if you add logging into the fetch function can you tell if the package configuration is definitely being loaded or not here? I'm also not sure I can see anything setting `configured`, so I'm wondering if this might be configuration precedence rather?

**Comment by AlbertWillemsen-Centric** [comment]
latest 0.20.18 version seems to fix this issue.
